pkg/gencosmosdb/cosmosdb/template.go

```diff
@@ -170,8 +187,11 @@ func (i *templateListIterator) Next(ctx context.Context) (templates *pkg.Templat
 	if i.continuation != "" {
 		headers.Set("X-Ms-Continuation", i.continuation)
 	}
+	if i.options != nil && len(i.options.PartitionKeyRangeID) > 0 {
```

Comment:
inclined to move this into setOptions and call setOptions from both NextRaw() functions